clippy_lints/src/types.rs

```diff
@@ -171,7 +171,7 @@ declare_clippy_lint! {
 
 declare_clippy_lint! {
     /// **What it does:** Checks for use of `&Box<T>` anywhere in the code.
-    /// Check the [Box documentation](https://doc.rust-lang.org/std/boxed/index.html) for more information.
+    /// Check the [`Box` documentation](std::boxed) for more information.
```

Comment:
Will these links also work on https://rust-lang.github.io/rust-clippy/?

Reply:
I'm not sure, how is that documentation built? If you're using mdbook then it won't work. I guess there's not much point to the PR then :/

Reply:
mdbook? You wish! We're still using our good ol' python script. 😄

Reply:
So yeah... I don't think we can change this in our lint documentation..